com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -48,62 +36,36 @@ internal enum __RpcExecStage
-#pragma warning restore IDE1006 // restore naming rule violation check
-        {
-            PooledNetworkWriter writer;
-
-            SetUpdateStage(ref serverRpcParams.Send);
-
-            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
-            {
-                throw new NotSupportedException(
-                    $"{nameof(NetworkUpdateStage.Initialization)} cannot be used as a target for processing RPCs.");
-            }
-
-            var messageQueueContainer = NetworkManager.MessageQueueContainer;
-            var networkDelivery = rpcDelivery == RpcDelivery.Reliable ? NetworkDelivery.ReliableSequenced : NetworkDelivery.UnreliableSequenced;
-
-            if (IsHost)
-            {
-                writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ServerRpc, Time.realtimeSinceStartup, networkDelivery,
-                    NetworkManager.ServerClientId, null, MessageQueueHistoryFrame.QueueFrameType.Inbound, serverRpcParams.Send.UpdateStage);
-            }
-            else
-            {
-                writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ServerRpc, Time.realtimeSinceStartup, networkDelivery,
-                    NetworkManager.ServerClientId, null, MessageQueueHistoryFrame.QueueFrameType.Outbound, NetworkUpdateStage.PostLateUpdate);
-
-                writer.WriteByte((byte)MessageQueueContainer.MessageType.ServerRpc);
-                writer.WriteByte((byte)serverRpcParams.Send.UpdateStage); // NetworkUpdateStage
-            }
 
-            writer.WriteUInt64Packed(NetworkObjectId); // NetworkObjectId
-            writer.WriteUInt16Packed(NetworkBehaviourId); // NetworkBehaviourId
-            writer.WriteUInt32Packed(rpcMethodId); // NetworkRpcMethodId
-
-
-            return writer.Serializer;
-        }
-
-#pragma warning disable IDE1006 // disable naming rule violation check
-        // RuntimeAccessModifiersILPP will make this `protected`
-        internal void __endSendServerRpc(NetworkSerializer serializer, uint rpcMethodId, ServerRpcParams serverRpcParams, RpcDelivery rpcDelivery)
-#pragma warning restore IDE1006 // restore naming rule violation check
+        public void SendServerRpc(ref FastBufferWriter writer, uint rpcMethodId, ServerRpcParams sendParams, RpcDelivery delivery)
         {
-            if (serializer == null)
-            {
-                return;
+            NetworkDelivery networkDelivery = NetworkDelivery.Reliable;
+            switch (delivery)
+            {
+                case RpcDelivery.Reliable:
+                    networkDelivery = NetworkDelivery.ReliableFragmentedSequenced;
+                    break;
+                case RpcDelivery.Unreliable:
+                    if (writer.Length > 1300 - sizeof(RpcMessage.RpcType) - sizeof(ulong) - sizeof(uint) - sizeof(ushort))
```

Comment:
I'd suggest to make this magic number a constant somewhere and reuse in other places.
Also instead of arbitrary 1300, what about an arbitrary flat 1024 (1kb)? :P

Reply:
Done.